File: wallendar-app/app/src/main/java/com/td/wallendar/utils/networking/RequestException.java
package com.td.wallendar.utils.networking;

public class RequestException extends Exception {

    private final int code;

    public RequestException(int code) {
        super("RequestException with code: " + code);
        this.code = code;
    }

    public RequestException(Throwable throwable) {
        super(throwable);
        this.code = -1;
    }

    public int getCode() {
        return code;
    }
}

File: wallendar-app/app/src/main/java/com/td/wallendar/utils/networking/RequestException.kt
package com.td.wallendar.utils.networking

class RequestException : Exception {
    private val code: Int

    constructor(code: Int) : super("RequestException with code: $code") {
        this.code = code
    }

    constructor(throwable: Throwable?) : super(throwable) {
        this.code = -1
    }

    fun getCode(): Int {
        return code
    }
}